core/src/main/scala/pureconfig/ConfigCursor.scala

```diff
@@ -103,27 +102,65 @@ sealed trait ConfigCursor {
    */
   def asCollectionCursor: Either[ConfigReaderFailures, Either[ConfigListCursor, ConfigObjectCursor]] = {
     if (isUndefined) {
-      fail(KeyNotFound(path, location, Set()))
+      failed(KeyNotFound.forKeys(path, Set()))
     } else {
       val listAtLeft = asListCursor.right.map(Left.apply)
       lazy val mapAtRight = asObjectCursor.right.map(Right.apply)
       listAtLeft
         .left.flatMap(_ => mapAtRight)
-        .left.flatMap(_ => fail(WrongType(value.valueType, Set(ConfigValueType.LIST, ConfigValueType.OBJECT), location, path)))
+        .left.flatMap(_ => failed(WrongType(value.valueType, Set(ConfigValueType.LIST, ConfigValueType.OBJECT))))
     }
   }
 
+  /**
+   * Returns a failed `ConfigReader` result resulting from scoping a `FailureReason` into the context of this cursor.
+   *
+   * This operation is the easiest way to return a failure from a `ConfigReader`.
+   *
+   * @param reason the reason of the failure
+   * @tparam A the returning type of the `ConfigReader`
+   * @return a failed `ConfigReader` result built by scoping `reason` into the context of this cursor.
+   */
+  def failed[A](reason: FailureReason): Either[ConfigReaderFailures, A] =
+    Left(ConfigReaderFailures(failureFor(reason)))
+
+  /**
+   * Returns a `ConfigReaderFailure` resulting from scoping a `FailureReason` into the context of this cursor.
+   *
+   * This operation is useful for constructing `ConfigReaderFailures` when there are multiple `FailureReason`s.
+   *
+   * @param reason the reason of the failure
+   * @return a `ConfigReaderFailure` built by scoping `reason` into the context of this cursor.
+   */
+  def failureFor(reason: FailureReason): ConfigReaderFailure =
+    ConvertFailure(reason, this)
+
+  /**
+   * Returns a failed `ConfigReader` result resulting from scoping a `Either[FailureReason, A]` into the context of
+   * this cursor.
+   *
+   * This operation is needed when control of the reading process is passed to a place without a `ConfigCursor`
+   * instance providing the nexessary context (for example, when `ConfigReader.fromString` is used. In those scenarios,
+   * the call should be wrapped in this method in order to turn `FailureReason` instances into `ConfigReaderFailures`.
+   *
+   * @param result the result of a config reading operation
+   * @tparam A the returning type of the `ConfigReader`
+   * @return a `ConfigReader` result built by scoping `reason` into the context of this cursor.
+   */
+  def scopeFailures[A](result: Either[FailureReason, A]): Either[ConfigReaderFailures, A] =
```

Comment:
I'm not sure about naming
- `scopeFailures` should be called `scopeFailure` imho because it's still one single `FailureReason`
- `failed` and `failureFor` names are strange when you compare them to `scopeFailures`: `scopeFailures` clearly tells the developer that its main task is "scoping" the given failure into the context of the cursor while `failed` and `failureFor` names don't contain scope at all. This is nitpicking :) but I find the names difficult to understand. I don't have a counterproposal though.

Reply:
I renamed `scopeFailures` to `scopeFailure` in ace155f, it makes more sense indeed. Regarding the others I am also not sure about the naming.

I think my rationale behind the names was that `failureFor´ was a simple constructor for a ConfigReaderFailure` from a `FailureReason`. Therefore, the focus of the method was not on scoping, but on the conversion.

`failed`, on the other hand, is a method targeted clearly at failure cases in `ConfigReader` implementations due to its signature, so I wanted to give a more fluent API (as in, calling `cur.failed` in a reader to signal a failed conversion).

`scopeFailures` is the only method of the three that doesn't necessarily result in a failed result. While `failed` always returns a `Left` and `failureFor` returns a `ConfigReaderFailure`, `scopeFailures` may return a `Right` result if its argument is not a failure.

I agree that the naming may be confusing, I'm open to ideas :)